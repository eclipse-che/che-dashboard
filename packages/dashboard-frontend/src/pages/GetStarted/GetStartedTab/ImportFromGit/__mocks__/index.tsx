/*
 * Copyright (c) 2018-2024 Red Hat, Inc.
 * This program and the accompanying materials are made
 * available under the terms of the Eclipse Public License 2.0
 * which is available at https://www.eclipse.org/legal/epl-2.0/
 *
 * SPDX-License-Identifier: EPL-2.0
 *
 * Contributors:
 *   Red Hat, Inc. - initial API and implementation
 */

import React from 'react';

import { Props } from '@/pages/GetStarted/GetStartedTab/ImportFromGit';

export class ImportFromGit extends React.PureComponent<Props> {
  public render() {
    const { hasSshKeys } = this.props;
    return (
      <div>
        <span data-testid="has-ssh-keys">{hasSshKeys ? 'true' : 'false'}</span>
      </div>
    );
  }
}
