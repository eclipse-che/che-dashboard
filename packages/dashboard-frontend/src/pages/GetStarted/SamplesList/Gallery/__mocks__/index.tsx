/*
 * Copyright (c) 2018-2024 Red Hat, Inc.
 * This program and the accompanying materials are made
 * available under the terms of the Eclipse Public License 2.0
 * which is available at https://www.eclipse.org/legal/epl-2.0/
 *
 * SPDX-License-Identifier: EPL-2.0
 *
 * Contributors:
 *   Red Hat, Inc. - initial API and implementation
 */

import React from 'react';

import { Props } from '@/pages/GetStarted/SamplesList/Gallery';

export default class SamplesListGallery extends React.PureComponent<Props> {
  render() {
    const { onCardClick, metadataFiltered = [] } = this.props;
    return (
      <div data-testid="samples-list-gallery">
        <div>Samples List Gallery</div>
        {metadataFiltered.map(metadata => (
          <div key={metadata.links.v2} data-testid="sample-card">
            {metadata.displayName}

            <button
              onClick={() => {
                onCardClick(metadata);
              }}
            >
              Select Sample
            </button>
          </div>
        ))}
      </div>
    );
  }
}
