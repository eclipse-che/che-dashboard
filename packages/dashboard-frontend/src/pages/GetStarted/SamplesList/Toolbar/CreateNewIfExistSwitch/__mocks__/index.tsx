/*
 * Copyright (c) 2018-2024 Red Hat, Inc.
 * This program and the accompanying materials are made
 * available under the terms of the Eclipse Public License 2.0
 * which is available at https://www.eclipse.org/legal/epl-2.0/
 *
 * SPDX-License-Identifier: EPL-2.0
 *
 * Contributors:
 *   Red Hat, Inc. - initial API and implementation
 */

import React from 'react';

import { Props } from '..';

export class CreateNewIfExistSwitch extends React.PureComponent<Props> {
  private isChecked = false;

  render(): React.ReactNode {
    return (
      <div data-testid="create-new-if-exist">
        {this.isChecked ? 'Create New If Existing On' : 'Create New If Existing Off'}
        <input
          data-testid="create-new-if-exist-switch"
          type="checkbox"
          onClick={() => {
            this.isChecked = !this.isChecked;
            this.props.onChange(this.isChecked);
          }}
          name="Toggle Create New If Existing"
        />
      </div>
    );
  }
}
