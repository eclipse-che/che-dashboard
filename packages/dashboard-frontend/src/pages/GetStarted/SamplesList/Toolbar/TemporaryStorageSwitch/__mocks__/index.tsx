/*
 * Copyright (c) 2018-2024 Red Hat, Inc.
 * This program and the accompanying materials are made
 * available under the terms of the Eclipse Public License 2.0
 * which is available at https://www.eclipse.org/legal/epl-2.0/
 *
 * SPDX-License-Identifier: EPL-2.0
 *
 * Contributors:
 *   Red Hat, Inc. - initial API and implementation
 */

import React from 'react';

import { Props } from '..';

export default class TemporaryStorageSwitch extends React.PureComponent<Props> {
  render(): React.ReactNode {
    const { isTemporary } = this.props;
    return (
      <div>
        {isTemporary ? 'Temporary Storage On' : 'Temporary Storage Off'}
        <input
          type="checkbox"
          onClick={() => this.props.onChange(!isTemporary)}
          name="Toggle Temporary Storage"
        />
      </div>
    );
  }
}
