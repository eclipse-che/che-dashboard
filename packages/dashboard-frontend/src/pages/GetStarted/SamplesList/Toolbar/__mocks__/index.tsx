/*
 * Copyright (c) 2018-2024 Red Hat, Inc.
 * This program and the accompanying materials are made
 * available under the terms of the Eclipse Public License 2.0
 * which is available at https://www.eclipse.org/legal/epl-2.0/
 *
 * SPDX-License-Identifier: EPL-2.0
 *
 * Contributors:
 *   Red Hat, Inc. - initial API and implementation
 */

import React from 'react';

import { Props } from '@/pages/GetStarted/SamplesList/Toolbar';

export default class SamplesListToolbar extends React.PureComponent<Props> {
  render(): React.ReactNode {
    const { isTemporary, onTemporaryStorageChange } = this.props;
    return (
      <div>
        <span>Samples List Toolbar</span>
        <span data-testid="isTemporary">{isTemporary ? 'true' : 'false'}</span>
        <button onClick={() => onTemporaryStorageChange(!isTemporary)}>Toggle isTemporary</button>
      </div>
    );
  }
}
