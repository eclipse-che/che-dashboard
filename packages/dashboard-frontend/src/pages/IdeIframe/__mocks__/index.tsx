/*
 * Copyright (c) 2018-2021 Red Hat, Inc.
 * This program and the accompanying materials are made
 * available under the terms of the Eclipse Public License 2.0
 * which is available at https://www.eclipse.org/legal/epl-2.0/
 *
 * SPDX-License-Identifier: EPL-2.0
 *
 * Contributors:
 *   Red Hat, Inc. - initial API and implementation
 */

import React from 'react';
import { Props } from '..';

export const iframeWorkspaceId = 'iframe-workspace-id';

export class IdeIframe extends React.Component<Props> {
  render(): React.ReactElement {
    return (
      <div data-testid="ide-iframe">
        <div data-testid="ideUrl">{this.props.ideUrl}</div>
        <div data-testid="isDevWorkspace">{this.props.isDevWorkspace}</div>
        <button onClick={() => this.props.onOpenWorkspacesList()}>Open Workspaces List</button>
        <button onClick={() => this.props.onWorkspaceRestartFromIframe(iframeWorkspaceId)}>
          Restart Workspace
        </button>
      </div>
    );
  }
}
