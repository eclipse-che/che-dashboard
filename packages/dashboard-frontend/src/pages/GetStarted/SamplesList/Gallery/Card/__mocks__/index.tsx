/*
 * Copyright (c) 2018-2024 Red Hat, Inc.
 * This program and the accompanying materials are made
 * available under the terms of the Eclipse Public License 2.0
 * which is available at https://www.eclipse.org/legal/epl-2.0/
 *
 * SPDX-License-Identifier: EPL-2.0
 *
 * Contributors:
 *   Red Hat, Inc. - initial API and implementation
 */

import React from 'react';

import { Props } from '@/pages/GetStarted/SamplesList/Gallery/Card';

export class SampleCard extends React.PureComponent<Props> {
  render(): React.ReactElement {
    const { metadata, onClick } = this.props;

    return (
      <div data-testid="sample-card">
        <div data-testid="sample-name">{metadata.displayName}</div>
        <button onClick={() => onClick()}>Select Sample</button>
      </div>
    );
  }
}
