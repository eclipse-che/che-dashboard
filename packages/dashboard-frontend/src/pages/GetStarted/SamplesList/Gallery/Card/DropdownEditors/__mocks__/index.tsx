/*
 * Copyright (c) 2018-2024 Red Hat, Inc.
 * This program and the accompanying materials are made
 * available under the terms of the Eclipse Public License 2.0
 * which is available at https://www.eclipse.org/legal/epl-2.0/
 *
 * SPDX-License-Identifier: EPL-2.0
 *
 * Contributors:
 *   Red Hat, Inc. - initial API and implementation
 */

import React from 'react';

import { Props } from '@/pages/GetStarted/SamplesList/Gallery/Card/DropdownEditors';

export class DropdownEditors extends React.PureComponent<Props> {
  render(): React.ReactNode {
    const { editors, onEditorSelect } = this.props;
    return (
      <div data-testid="card-actions-dropdown">
        <div>spanDropdownEditors</div>
        {editors.map(editor => (
          <button
            data-testid="card-action"
            key={editor.id}
            onClick={() => onEditorSelect(editor.id)}
          >
            {editor.name}
          </button>
        ))}
        <button onClick={() => onEditorSelect(editors[0].id)}>Select Editor</button>
      </div>
    );
  }
}
