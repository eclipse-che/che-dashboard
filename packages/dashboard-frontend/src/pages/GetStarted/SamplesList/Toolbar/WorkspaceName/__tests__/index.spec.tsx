/*
 * Copyright (c) 2018-2024 Red Hat, Inc.
 * This program and the accompanying materials are made
 * available under the terms of the Eclipse Public License 2.0
 * which is available at https://www.eclipse.org/legal/epl-2.0/
 *
 * SPDX-License-Identifier: EPL-2.0
 *
 * Contributors:
 *   Red Hat, Inc. - initial API and implementation
 */

import { screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import React from 'react';
import { Provider } from 'react-redux';
import { Store } from 'redux';

import WorkspaceNameFormGroup from '@/pages/GetStarted/SamplesList/Toolbar/WorkspaceName';
import getComponentRenderer from '@/services/__mocks__/getComponentRenderer';
import { DevWorkspaceBuilder } from '@/store/__mocks__/devWorkspaceBuilder';
import { MockStoreBuilder } from '@/store/__mocks__/mockStore';

const { renderComponent, createSnapshot } = getComponentRenderer(getComponent);

const mockOnChange = jest.fn();

// mute console.error
console.error = jest.fn();

describe('Workspace Name', () => {
  let store: Store;

  beforeEach(() => {
    const workspaces = [0, 1, 2, 3, 4].map(i =>
      new DevWorkspaceBuilder()
        .withId('wksp-' + i)
        .withName('wksp-' + i)
        .build(),
    );
    store = new MockStoreBuilder().withDevWorkspaces({ workspaces }).build();
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  test('snapshot', () => {
    const snapshot = createSnapshot(store);
    expect(snapshot.toJSON()).toMatchSnapshot();
  });

  describe('valid value', () => {
    test('default empty value', () => {
      renderComponent(store);

      const input = screen.getByRole('textbox');

      expect(input).toHaveValue('');
    });

    test('change to valid value', async () => {
      renderComponent(store);

      const input = screen.getByRole('textbox');

      const workspaceName = 'new-workspace';
      await userEvent.click(input);
      await userEvent.paste(workspaceName);

      expect(mockOnChange).toHaveBeenNthCalledWith(1, workspaceName);
    });

    test('clean up value', async () => {
      renderComponent(store);

      const input = screen.getByRole('textbox');

      const workspaceName = 'new-workspace';
      await userEvent.click(input);
      await userEvent.paste(workspaceName);
      expect(mockOnChange).toHaveBeenNthCalledWith(1, workspaceName);

      await userEvent.clear(input);
      expect(mockOnChange).toHaveBeenNthCalledWith(2, '');
    });
  });
  describe('invalid value', () => {
    test('used workspace name value', async () => {
      const workspaceName = 'new-workspace';
      const callbacks: { reset?: () => void } = {};

      renderComponent(store, callbacks);

      const input = screen.getByRole('textbox');

      await userEvent.click(input);
      await userEvent.paste(workspaceName);

      expect(mockOnChange).toHaveBeenNthCalledWith(1, workspaceName);
      mockOnChange.mockReset();

      callbacks.reset?.();

      expect(mockOnChange).toHaveBeenNthCalledWith(1, '');
      mockOnChange.mockReset();

      await userEvent.click(input);
      await userEvent.paste(workspaceName);

      expect(mockOnChange).toHaveBeenNthCalledWith(1, '');
      mockOnChange.mockReset();

      await userEvent.paste('-new');

      expect(mockOnChange).toHaveBeenNthCalledWith(1, `${workspaceName}-new`);
    });
    test('change to too long value', async () => {
      renderComponent(store);

      const input = screen.getByRole('textbox');

      const workspaceName = 'a'.repeat(64);
      await userEvent.click(input);
      await userEvent.paste(workspaceName);

      expect(mockOnChange).toHaveBeenNthCalledWith(1, '');
    });
    test('change to invalid pattern value', async () => {
      renderComponent(store);

      const input = screen.getByRole('textbox');

      const workspaceName = 'invalid name!';
      await userEvent.click(input);
      await userEvent.paste(workspaceName);

      expect(mockOnChange).toHaveBeenNthCalledWith(1, '');
    });
    test('change to existing name value', async () => {
      renderComponent(store);

      const input = screen.getByRole('textbox');

      const workspaceName = 'wksp-1';
      await userEvent.click(input);
      await userEvent.paste(workspaceName);

      expect(mockOnChange).toHaveBeenNthCalledWith(1, '');
    });
  });
});

function getComponent(store: Store, callbacks?: { reset?: () => void }) {
  return (
    <Provider store={store}>
      <WorkspaceNameFormGroup
        onChange={workspaceName => mockOnChange(workspaceName)}
        callbacks={callbacks}
      ></WorkspaceNameFormGroup>
    </Provider>
  );
}
