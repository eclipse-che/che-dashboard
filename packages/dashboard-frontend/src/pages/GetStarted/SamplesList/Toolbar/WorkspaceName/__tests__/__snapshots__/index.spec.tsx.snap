// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Workspace Name snapshot 1`] = `
<div
  className="pf-c-form__group inputFormGroup"
>
  <div
    className="pf-c-form__group-control"
  >
    <input
      aria-invalid={false}
      aria-label={null}
      className="pf-c-form-control"
      data-ouia-component-id="OUIA-Generated-TextInputBase-1"
      data-ouia-component-type="PF4/TextInput"
      data-ouia-safe={true}
      disabled={false}
      id="wrks-name-input"
      maxLength={64}
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      placeholder="Enter a name for the sample workspace (Optional)"
      required={false}
      type="text"
      value=""
    />
  </div>
</div>
`;
